### ISSUE TITLE
onewire iButton CONFIGURATION_ERROR - required properties missing

### ISSUE THREAD

**Issue by UrsusS** [openhab/openhab-addons#14066]
I have a couple of onewire iButton keys (DS2401). They are used to set an operational mode for OH (e.g. vacation mode)
They are configured as onewire basic things
e.g. 
```
Bridge onewire:owserver:Flur	"OneWire Bridge" @ "Flur EG"
[
	network-address="127.0.0.1"
]
{
	Thing basic iButton_Haus "Schlüssel Haus" @ "Flur EG"			// lila
	[
		id="01.7ECCxxxx0000",
		refresh=3
	]
	{
		Channels:
			Type present : Present
	}
}
```
They are all configured in the same way, just the ID is different

If the key is not present, the thing shows "OFFLINE". I assume that as a system behaviour, eventhough the present channel could cover that.
But what I'm wondering is that in case of the thing in OFFLINE mode, it shows eighter a "CONFIGURATION_ERROR" with "required properties missing" as description or a "COMMUNICATION_ERROR" with "slave missing" as description.
The configuration error is, when the iButton hasn't been online since the last restart of OH. When the iButton has been there, the thing properties "modelId" and "vendor" are available and set. 
If the missing properties are leading to an error, it should be possible to set the properties already in the thing configuration. They are ignored, if they are provided in the textual configuration.

OpenHAB 3.4.0, but also in previous versions

**Comment by lolodomo** [comment]
I just take a very quick look to the code.

"CONFIGURATION_ERROR/required properties missing" is set apparently until the bridge handler retrieves some data that will set properties on the thing. Then the thing handler is initialized by the the bridge handler by calling initialize() ! All this seems to be because the thing handler needs to know if the sensor type (coming for the MODELID property is supported or not.
This looks like a strange conception of bridge/child handlers.

"COMMUNICATION_ERROR/slave missing" is apparently set when the presence state is OFF.

**Comment by J-N-K** [comment]
Simple solution: Don't use text files. 

UI configured things keep their properties, file configured don't. TTBOMK there is no way to configure properties in text files, but that is not a limitation of the binding but a limitation of the DSL.

`OFFLINE/COMMUNICATION_ERROR` is the correct state, because no communication is possible when the iButton is not attached to the bus. 

@lolodomo: We could move the `initialize` call to `OwBaseThingHandler.updateSensorProperties` but that technically doesn't make a difference. This is necessary because we need to synchronize the calls to the owserver instance. 

Calls to get the sensor information will timeout when the sensor is not connected (which is often the case for iButtons) and that is quite time-consuming (typically some seconds). Just queuing the requests to the server would result in possible long delays for the refresh of sensor data. We mitigate that by maintaining a special queue for the property updates and only call one of them between regular refreshes. If the update fails, we re-add it to the end of the queue. So what the bridge handler really does is load balancing by postponing property updates for non-initialized things. After the update the thing handler needs to re-initialized manually, because there is no automatic `dispose()/initialize()` after a property update.

**Comment by lsiepel** [comment]
> Simple solution: Don't use text files.
> 
> UI configured things keep their properties, file configured don't. TTBOMK there is no way to configure properties in text files, but that is not a limitation of the binding but a limitation of the DSL.
> 
> `OFFLINE/COMMUNICATION_ERROR` is the correct state, because no communication is possible when the iButton is not attached to the bus.
> 
> @lolodomo: We could move the `initialize` call to `OwBaseThingHandler.updateSensorProperties` but that technically doesn't make a difference. This is necessary because we need to synchronize the calls to the owserver instance.
> 
> Calls to get the sensor information will timeout when the sensor is not connected (which is often the case for iButtons) and that is quite time-consuming (typically some seconds). Just queuing the requests to the server would result in possible long delays for the refresh of sensor data. We mitigate that by maintaining a special queue for the property updates and only call one of them between regular refreshes. If the update fails, we re-add it to the end of the queue. So what the bridge handler really does is load balancing by postponing property updates for non-initialized things. After the update the thing handler needs to re-initialized manually, because there is no automatic `dispose()/initialize()` after a property update.

What about a configuration option that when empty/not set has the current behaviour. And when set it overwrites the `MODELID` property?

It would need you to know the models upfront, not sure if that is possible at all.  Text based config would then be fully functional.

**Comment by J-N-K** [comment]
I don't think we should start adding workarounds for core issues in add-ons. Someone who thinks textual config is a valuable part of openHAB should fix it in core.

There is no malfunction in the binding and except a different `ThingStatusDetail` the thing behaves exactly the same, even with the missing property. IMO this all behaves exactly as it should: missing properties ARE a configuration error, that is fixed once the thing is appearing on the bus for the first time.

**Comment by lsiepel** [comment]
Another argument could be that the prior PR (by relying on property’s the way it does) lacks text config support and should have been designed different. But it is what it is.

One last thing that comes to mind is that regardless of the configuration method, the modelid could be stored in a JSON file in the userdata folder. 

As you said, fixing it in core with the property related to the text config seems not feasable due to the limitations. So that is a dead end.

**Comment by J-N-K** [comment]
As I said: it works perfectly fine. The ONLY difference is that a device that is not on the bus when openHAB starts up is shown as `OFFLINE/CONFIGURATION_ERROR` until it connects for the first time. Then the properties are set and there is NO difference anymore. 

It does not lack textual support in any way. I can't think of a use-case where a different thing status detail would be useful except for debugging purposes.

**Comment by lsiepel** [comment]
I should rephrase; it doesn’t’ lack support, the experience is just a little less.  Agree that it is overkill.

what you say makes totally sense, still looking to make this clear to users if the binding.

To prevent questions would it be possible to adjust the status detail text? ATM it looks fatal, but it was just not yet determined.
Maybe “required properties could not (yet) be determined”
